### PR TITLE
Fixed PermissionError when trying to delete extracted frames by removing files' read-only permissions

### PR DESCRIPTION
**Pull request by zxsean**
Try to fix this issue:
https://github.com/k4yt3x/video2x/issues/321

### CODE REVIEW

File: src/upscaler.py [hunk 1]
```diff
@@ -703,7 +706,17 @@ def run(self):
 
                 # downscale frames with Lanczos
                 Avalon.info(_('Lanczos downscaling frames'))
-                shutil.rmtree(self.extracted_frames)
+                try:
+                    shutil.rmtree(self.extracted_frames)
+                except:
```
Comment (k4yt3x) [inline src/upscaler.py:711]: Please do not use bare `except`

File: src/upscaler.py [hunk 2]
```diff
@@ -703,7 +706,17 @@ def run(self):
...
+                    if platform.system() == 'Windows':
+                        Avalon.info(_("Change file's attribute."))
```
Comment (k4yt3x) [inline src/upscaler.py:713]: This line is not clear. What file's what attribute are you changing?
Also, there should be no periods at the end of command line outputs.

File: src/upscaler.py [hunk 3]
```diff
@@ -703,7 +706,17 @@ def run(self):
...
+                        for root, dirs, files in os.walk(self.extracted_frames):
```
Comment (k4yt3x) [inline src/upscaler.py:714]: Do not use `os` for file operations. Use `pathlib.Path`.

**Review by k4yt3x — CHANGES_REQUESTED** [review]

**Comment by zxsean** [comment]
ok~ thx i will fix this~

File: src/upscaler.py [hunk 4]
```diff
@@ -703,7 +706,16 @@ def run(self):
 
                 # downscale frames with Lanczos
                 Avalon.info(_('Lanczos downscaling frames'))
-                shutil.rmtree(self.extracted_frames)
+                try:
+                    shutil.rmtree(self.extracted_frames)
+                except Exception:
```
Comment (k4yt3x) [inline src/upscaler.py:711]: Exception is too broad. You might be catching that's not intended to be caught here.
Use `except PermissionError` to be specific.

File: src/upscaler.py [hunk 5]
```diff
@@ -703,7 +706,16 @@ def run(self):
...
+                    if platform.system() == 'Windows':
+                        for file in self.extracted_frames.iterdir():
+                            Avalon.info(_("Remove file:[%s] readOnly attribute." % file))
```
Comment (k4yt3x) [inline src/upscaler.py:714]: Formatting should happen outside of the gettext function. Also, `%` operator is an outdated method. Use `.format`.
Also, please be consistent with your use of quotation marks. This project uses single quotes.
e.g., `Avalon.info(_('Removing read-only attribute from file {}').format(file))`

File: src/upscaler.py [hunk 6]
```diff
@@ -703,7 +706,16 @@ def run(self):
 
                 # downscale frames with Lanczos
                 Avalon.info(_('Lanczos downscaling frames'))
-                shutil.rmtree(self.extracted_frames)
+                try:
+                    shutil.rmtree(self.extracted_frames)
+                except Exception:
+                    if platform.system() == 'Windows':
+                        for file in self.extracted_frames.iterdir():
```
Comment (k4yt3x) [inline src/upscaler.py:713]: Historically, `file` was a keyword, an alias for `open.
I would recommend using something more specific like `extracted_frame`.

File: src/upscaler.py [hunk 7]
```diff
@@ -703,7 +706,16 @@ def run(self):
...
+                            Avalon.info(_("Remove file:[%s] readOnly attribute." % file))
+                            os.chmod(file, stat.S_IWRITE)
```
Comment (k4yt3x) [inline src/upscaler.py:715]: This is a file object operation, so it should be handled by the file object.
Use `extracted_frame.chmod`.

File: src/upscaler.py [hunk 8]
```diff
@@ -35,6 +35,9 @@
 import tempfile
 import time
 import traceback
+import os
```
Comment (k4yt3x) [inline src/upscaler.py:38]: Remove this redundant import.

File: src/upscaler.py (hunk 5 above)
Comment (k4yt3x) [reply src/upscaler.py:714]: Also are you sure that you'll want to print this out?
Even if you do, it shouldn't be `Avalon.info` but `Avalon.debug_info`.
This shouldn't have a vibrant color and get outputted to stdout.

File: src/upscaler.py [hunk 9]
```diff
@@ -703,7 +706,16 @@ def run(self):
...
-                shutil.rmtree(self.extracted_frames)
+                try:
+                    shutil.rmtree(self.extracted_frames)
+                except Exception:
+                    if platform.system() == 'Windows':
+                        for file in self.extracted_frames.iterdir():
+                            Avalon.info(_("Remove file:[%s] readOnly attribute." % file))
+                            os.chmod(file, stat.S_IWRITE)
+
+                        shutil.rmtree(self.extracted_frames)
```
Comment (k4yt3x) [inline src/upscaler.py:717]: What if it's not windows? You need to throw the exception again, otherwise it will keep on going ignoring the errors.

**Review by k4yt3x — CHANGES_REQUESTED** [review]

**Comment by k4yt3x** [comment]
Your commit messages need to be more meaningful as well.

**Comment by k4yt3x** [comment]
I'm hesitating to merge this PR. If a user has set read-only permissions on a file or folder/directory, program should be expected to respect those permissions instead of altering them without notifying the user.

**Comment by k4yt3x** [comment]
@RhynarAI please take a look at this.

**Comment by zxsean** [comment]
Not particularly good at python~~233.
i will fix these.

**Comment by k4yt3x** [comment]
@zxsean you can wait until @jekabsmilbrets provides the necessary supplemental information.
@RhynarAI also needs to review this.

**Comment by zxsean** [comment]
Sure.

**Comment by RhynarAI** [comment]
What I am thinkin of about is: How did the files get  the "read only" attribute in the first place? A temp folder for extracted frames should not have set these permissions by default, unless someone has weird user restrictions on certain directories, in which case he should change these user rights or use another directory..

**Comment by jekabsmilbrets** [comment]
I tested on version 4.7.0 and it works perfectly!

**Comment by k4yt3x** [comment]
@jekabsmilbrets Excellent.

I am closing this PR due to the following considerations:

- There is no evidence that Video2X or its components will change directory/file permissions. Therefore, there's no need altering any permissions.
- Default permissions and attributes of files and directories should be respected. An error should be thrown when the files/directories cannot be read/written instead of changing the file/directory attributes.

Thank you for opening this PR. Please leave additional comments if you'd like to.

**Pull request closed without merging**